### ISSUE TITLE
BUG: GMM t_test

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#2446]
using a plain GMM, I get an exception in t_test

```
>>> print(res_gmm2.t_test(constraint))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "m:\josef_new\eclipse_ws\statsmodels\statsmodels_py34\statsmodels\base\model.py", line 1202, in t_test
    r_matrix=r_matrix, cov_p=cov_p)))
  File "m:\josef_new\eclipse_ws\statsmodels\statsmodels_py34\statsmodels\sandbox\regression\gmm.py", line 1144, in cov_params
    covparams = self.calc_cov_params(moms, gradmoms, **kwds)
TypeError: calc_cov_params() got an unexpected keyword argument 'cov_p'
>>> 
```

**Comment by josef-pkt** [comment]
maybe for 0.7 still

started here https://github.com/statsmodels/statsmodels/compare/master...josef-pkt:fix_gmm
but it's not correct yet, failed in treatment/IPW branch on TravisCI

**Comment by josef-pkt** [comment]
GMM doesn't seem to have unit tests for pvalues. They are printed by `summary` so they will be available through inheritance.  adding `use_t` either True or False doesn't make the unit tests fail.

(IV2SLS has test coverage for pvalues and t_test)

**Comment by josef-pkt** [comment]
fixed in #2497
